admin/admin-filters-columns.php

```diff
@@ -266,101 +245,80 @@ public function add_term_column( $columns ) {
 			return $out;
 		}
 
-		$term_id = (int) $term_id;
 		$language = $this->model->get_language( substr( $column, 9 ) );
 
 		if ( empty( $language ) ) {
 			return $out;
 		}
 
-		// Link to edit term ( or a translation )
-		if ( ( $id = $this->model->term->get( $term_id, $language ) ) && $term = get_term( $id, $taxonomy ) ) {
-			if ( $term instanceof WP_Term && $link = get_edit_term_link( $id, $taxonomy, $post_type ) ) {
-				$flag = '';
-				if ( $id === $term_id ) {
-					$flag = $this->get_flag_html( $language );
-					$class = 'pll_column_flag';
-					/* translators: accessibility text, %s is a native language name */
-					$s = sprintf( __( 'Edit this item in %s', 'polylang' ), $language->name );
-				} else {
-					$class = esc_attr( 'pll_icon_edit translation_' . $id );
-					/* translators: accessibility text, %s is a native language name */
-					$s = sprintf( __( 'Edit the translation in %s', 'polylang' ), $language->name );
-				}
-				$out .= sprintf(
-					'<a class="%1$s" title="%2$s" href="%3$s"><span class="screen-reader-text">%4$s</span>%5$s</a>',
-					$class,
-					esc_attr( $term->name ),
-					esc_url( $link ),
-					esc_html( $s ),
-					$flag
-				);
-			} elseif ( $id === $term_id ) {
-				$out .= sprintf(
-					'<span class="pll_column_flag"><span class="screen-reader-text">%1$s</span>%2$s</span>',
-					/* translators: accessibility text, %s is a native language name */
-					esc_html( sprintf( __( 'This item is in %s', 'polylang' ), $language->name ) ),
-					$this->get_flag_html( $language )
-				);
-			}
-		}
+		$tr_id   = $this->model->term->get( $term_id, $language );
+		$tr_term = $tr_id ? get_term( $tr_id, $taxonomy ) : false;
 
-		// Link to add a new translation
-		else {
-			$out .= $this->links->new_term_translation_link( $term_id, $taxonomy, $post_type, $language );
+		if ( ! $tr_term instanceof WP_Term ) {
+			// Link to add a new translation: no translation for this language yet, or it doesn't exist anymore.
+			return $out . $this->links->new_term_translation_link( $term_id, $taxonomy, $post_type, $language );
 		}
 
-		if ( $this->get_first_language_column() === $column ) {
-			$out .= sprintf( '<div class="hidden" id="lang_%d">%s</div>', intval( $term_id ), esc_html( $lang->slug ) );
-
-			/**
-			 * Filters the output of the first language column in the terms list table.
-			 *
-			 * @since 3.7
-			 *
-			 * @param string $output  First language column output.
-			 * @param int    $term_id Term ID.
-			 * @param string $lang    Language code.
-			 */
-			$out = apply_filters( 'pll_first_language_term_column', $out, $term_id, $lang->slug );
+		// Link to edit (or not) the term or a translation.
+		$url  = $this->links->get_edit_term_translation_link( $tr_term->term_id, $taxonomy, $post_type, $language );
+		$out .= $this->links->get_item_edition_link( $url, $tr_term->term_id, $tr_term->name, $language, $tr_term->term_id === $term_id ? 'flag' : 'icon' );
```

Comment:
Same as above, I poorly described my suggestion but I was thinking of a single method.